### DISCUSSION TITLE
Improve `useState` typing

### DISCUSSION THREAD

**Discussion by nandi95** [nuxt/nuxt#18191] [Ideas]
There's already an option in vue 3 that is normlly used in conjution with `provide` and `inject`: `InjectionKey`.
It looks something like this `export const loaderKey: InjectionKey<Ref<InstanceType<typeof Loader>>> = Symbol('loader');` and when this `loaderKey` passed to the `inject` it will infer the type from the injection key.
Can the same be done with `useState`? especially that the key should be unique, symbol is actually fits the job.

**Comment by pi0** [comment]
Seems a nice idea!! Do you mind making a PR @nandi95?

**Reply by nandi95** [reply]
I'll make one when I have time asap.

**Comment by nandi95** [comment]
I guess something similar can be done with [useAsyncData](https://v3.nuxtjs.org/docs/usage/data-fetching#useasyncdata) as that also requires a unique key?
@danielroe
